File: src/test/java/com/frightanic/smn/api/geojson/CrsTypeTest.java
package com.frightanic.smn.api.geojson;

import org.junit.Test;

import static org.hamcrest.core.Is.is;
import static org.junit.Assert.assertThat;

/**
 * Tests CrsType.
 */
public class CrsTypeTest {

  /**
   * See method name.
   */
  @Test
  public void shouldHaveOgnUrn4326ForWsg84() {
    // given
    // when
    // then
    assertThat(CrsType.WGS84.getUrn(), is("urn:ogc:def:crs:EPSG::4326"));
  }

  /**
   * See method name.
   */
  @Test
  public void shouldHaveOgnUrn21781ForCH1903() {
    // given
    // when
    // then
    assertThat(CrsType.CH1903.getUrn(), is("urn:ogc:def:crs:EPSG::21781"));
  }
}

File: src/test/kotlin/com/frightanic/smn/api/geojson/CrsTypeTest.kt
package com.frightanic.smn.api.geojson

import com.frightanic.smn.api.geojson.CrsType.CH1903
import com.frightanic.smn.api.geojson.CrsType.WGS84

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.hamcrest.core.Is
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

/**
 * Tests CrsType.
 */
internal class CrsTypeTest {
    @Test
    fun `should have OGN URN 4326 for WGS84`() {
        // given
        // when
        // then
        assertThat(WGS84.urn, `is`("urn:ogc:def:crs:EPSG::4326"))
    }

    @Test
    fun `should have OGN URN 21781 for CH1903`() {
        // given
        // when
        // then
        assertThat(CH1903.urn, `is`("urn:ogc:def:crs:EPSG::21781"))
    }
}
